RHS HIDF Template

ante185 opened the pull request:
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
Adds the RHS HIDF faction.
This template challenges players to a fight against an foe with an outdated cold war era army, but friends in high places allow them some aces up their sleeve.

this also adds a config fix for the RHS M3A1 gun, as its lack of AI firemodes restricted its usefulness to none

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:
Dependencies: 
This template depends on #3172 as it allows their CAS plane to work.

Tiny-DM commented on A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:
```diff
@@ -0,0 +1,856 @@
+["vehiclesHelisLightAttack", ["rhs_uh1h_hidf_gunship"]] call _fnc_saveToTemplate; 
+["vehiclesHelisAttack", []] call _fnc_saveToTemplate;
+
+["vehiclesArtillery", ["RHS_M119_WD"]] call _fnc_saveToTemplate;        // "rhsusf_M142_usarmy_D" - OP until we can customize magazines
```
remove comment

Tiny-DM commented on A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:
```diff
@@ -0,0 +1,856 @@
+
+["vehiclesArtillery", ["RHS_M119_WD"]] call _fnc_saveToTemplate;        // "rhsusf_M142_usarmy_D" - OP until we can customize magazines
+["magazines", createHashMapFromArray [
+["rhsusf_m109_usarmy", ["rhs_mag_155mm_m795_28"]],
```
remove unnecessary ammo entries

Tiny-DM commented on A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:
```diff
@@ -0,0 +1,856 @@
+
+//Config special vehicles
+["vehiclesMilitiaLightArmed", ["rhsgref_hidf_m1025_m2"]] call _fnc_saveToTemplate;
+["vehiclesMilitiaTrucks", ["rhs_zil131_msv"]] call _fnc_saveToTemplate;
```
Eastern truck stands out in a template of US surplus

Tiny-DM commented on A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:
```diff
@@ -0,0 +1,856 @@
+_loadoutData set ["sniperRifles", []];
+
+_loadoutData set ["lightATLaunchers", ["rhs_weap_m72a7"]];
+_loadoutData set ["ATLaunchers", [
```
Less optics maybe? The faction is supposed to be really poor.

Tiny-DM commented on A3A/addons/core/Templates/Templates/RHS/RHS_AI_HIDF.sqf:
```diff
@@ -0,0 +1,856 @@
+]];
+_sfLoadoutData set ["SMGs", [
+["rhs_weap_m3a1_specops", "", "", "rhsusf_acc_compm4", ["rhsgref_30rnd_1143x23_M1911B_SMG"], [], ""],
+["rhs_weap_m4a1_carryhandle", "rhsusf_acc_rotex5_grey", "acc_flashlight", "", ["rhs_mag_30Rnd_556x45_M855A1_Stanag"], [], ""]
```
An M4 in the SMGs?

ante185 replied:
yep

Tiny-DM reviewed (approved):
lgtm

ante185 commented:
Update:
Replaced the modern US trucks with trucks of a more suitable vintage.
Created olive green versions of the TOW hmmwv, m113 m240, m113 MEV, and M2A2, added to template
